app/src/main/java/io/gnosis/safe/ui/transactions/details/TransactionDetailsViewModel.kt

```diff
@@ -34,6 +38,19 @@ class TransactionDetailsViewModel
                 true == ownerCredentialsRepository.retrieveCredentials()?.let { credentials ->
             executionInfo.signers.contains(credentials.address) && !executionInfo.confirmations.map { it.signer }.contains(credentials.address)
         }
+
+    suspend fun validateSafeTxHash(transaction: TransactionDetails): Boolean {
+        return kotlin.runCatching {
+            val safe = safeRepository.getActiveSafe()
+            val safeTxHash = (transaction.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails).safeTxHash
+            val calculatedSafeTxHash = calculateSafeTxHash(safe!!.address, transaction)?.toHexString()?.addHexPrefix()
+            safeTxHash == calculatedSafeTxHash
+        }.onSuccess {
+            return it
+        }.onFailure {
+            return false
+        }.getOrDefault(false)
```

Comment:
If you use `getOrDefault` you don't need to have `onSuccess` and `onFailure` at all. Just return what `getOrDefault` returns.

Reply:
```suggestion
return kotlin.runCatching {
            val safe = safeRepository.getActiveSafe()
            val safeTxHash = (transaction.detailedExecutionInfo as DetailedExecutionInfo.MultisigExecutionDetails).safeTxHash
            val calculatedSafeTxHash = calculateSafeTxHash(safe!!.address, transaction)?.toHexString()?.addHexPrefix()
            safeTxHash == calculatedSafeTxHash
        }.getOrDefault(false)
```